src/librustc_trans/mir/mod.rs

```diff
@@ -176,7 +184,26 @@ impl<'tcx> LocalRef<'tcx> {
 
 ///////////////////////////////////////////////////////////////////////////
 
-pub fn trans_mir<'a, 'tcx: 'a>(fcx: &'a FunctionContext<'a, 'tcx>, mir: &'a Mir<'tcx>) {
+pub fn trans_mir<'a, 'tcx: 'a>(
+    fcx: &'a FunctionContext<'a, 'tcx>,
+    mir: &'a Mir<'tcx>,
+    instance: Instance<'tcx>,
+    sig: &ty::FnSig<'tcx>,
+    abi: Abi,
+) {
+    let def_id = instance.def;
+    let local_id = fcx.ccx.tcx().map.as_local_node_id(def_id);
+    let no_debug = if let Some(id) = local_id {
+        fcx.ccx.tcx().map.attrs(id).iter().any(|item| item.check_name("no_debug"))
+    } else {
+        fcx.ccx.sess().cstore.item_attrs(def_id).iter().any(|item| item.check_name("no_debug"))
```

Comment:
This is silly, `tcx` has a method to get attributes which should work regardless of local/non-local. I think.